### PR TITLE
refactor 'Community Guideline' chapter

### LINKED ISSUE #124: Refactor "Community Guideline"

### PR DESCRIPTION
**Pull request by zhangyangyu**
### What issue does this PR solve?

- close #124 

### What is changed:

Aims to provide readers the info about CoC and governance. Besides, try to give a welcome tone and emphasize any contribution in any kinds are all appreciated in the community.

### CODE REVIEW

**Comment by zhangyangyu** [comment]
One thing I'm not sure is whether just link "TOC" and "teams" info in the governance part or try to embed their content totally in the dev guide.

File: src/contribute-to-tidb/community-guideline.md [hunk 1]
```diff
@@ -1,31 +1,33 @@
...
 
-## General Development Process
+* Write Code
+* Increase test coverage
+* Improve documentation
+* Review and comment on pull requests from other developers
+* Report an issue
+* Triage issues on GitHub
+* Participate in discussion online or offline about TiDB technolegies
+* Answer questions on Stack Overflow, write blogs about TiDB and more
```
Comment (zhangyangyu) [inline src/contribute-to-tidb/community-guideline.md:12]: Hmm, this part seems duplicated with the main chapter.

**Review by xxchan — COMMENTED** [review]
As https://github.com/pingcap/tidb-dev-guide/issues/122#issuecomment-913392087, I think we should include all information in https://github.com/pingcap/community/blob/master/GOVERNANCE.md (like detailed role introductions)?

**Comment by tisonkun** [comment]
> As [#122 (comment)](https://github.com/pingcap/tidb-dev-guide/issues/122#issuecomment-913392087), I think we should include all information in https://github.com/pingcap/community/blob/master/GOVERNANCE.md (like detailed role introductions)?

You mistake the sentence. Please read it again.

File: src/contribute-to-tidb/community-guideline.md [hunk 2]
```diff
@@ -1,31 +1,24 @@
 # Community Guideline
 
-TiDB governs by meritocracy and adopts a model consists of reviewer, committer, and maintainer. We believe that it is important to create an inclusive community where everyone can use, contribute to, and influence the direction of the project. See [the contributors page](https://github.com/pingcap/tidb/graphs/contributors) for the current list of contributors.
+TiDB community aims to provide harassment-free, welcome and friendly experience for everyone. The first and most important thing for any participant in the community is be friendly and repectful to others. Improper behaviour will be warned and punished.
```
Comment (tisonkun) [inline src/contribute-to-tidb/community-guideline.md:3]:
```suggestion
TiDB community aims to provide harassment-free, welcome and friendly experience for everyone. The first and most important thing for any participant in the community is be friendly and respectful to others. Improper behaviors will be warned and punished.
```

File: src/contribute-to-tidb/community-guideline.md [hunk 3]
```diff
@@ -1,31 +1,24 @@
...
-## General Development Process
+We appreciate any contribution in any form to TiDB community. Thanks so much for your interest and enthusiasm on TiDB!
 
-Everyone in the community is welcomed to send patches, documents, and propose new directions to the project. The key guideline here is to enable everyone in the community to get involved and participate the decision and development. When major changes are proposed, a design document should be sent to allow discussion by the community. We encourage public discussion, archivable channels such as GitHub issues and the [TiDB Internals forum](https://internals.tidb.io/), so that everyone in the community can participate and review the process later.
+## Code of Conduct
 
-Code reviews are one of the key ways to ensure the quality of the code. High-quality code reviews prevent technical debt for the long term and are crucial to the success of the project.
+TiDB community refuses any kind of harmful behaviour to the community or community members. Everyone should read the [CoC](https://github.com/pingcap/community/blob/master/CODE_OF_CONDUCT.md?from=from_parent_mindnote) and ensure proper behaviour while participating in the community.
```
Comment (tisonkun) [inline src/contribute-to-tidb/community-guideline.md:9]:
```suggestion
TiDB community refuses any kind of harmful behavior to the community or community members. Everyone should read our [Code of Conduct](https://github.com/pingcap/community/blob/master/CODE_OF_CONDUCT.md) and keep proper behavior while participating in the community.
```

File: src/contribute-to-tidb/community-guideline.md [hunk 4]
```diff
@@ -1,31 +1,24 @@
...
+* [teams](https://github.com/pingcap/community/tree/master/teams): Teams are persistent open groups that focus on a part of the TiDB projects. A team has its reviewer, committer and maintainer, and owns one or more repositories. Team level decision making comes from its maintainers.
 
-## Reviewers
+A typical promoted path for a TiDB developer is from user to reviewer, then committer and maintainer, finally maybe a TOC member.  But gaining more roles doesn't mean you have any priviledge over other community members or even any right to control them. Everyone in TiDB community are equal and share the responsibility to collaborate constructively with other contributors, building a friendly community. The roles are a natural reward for your substantial contribution in TiDB development and provide you more rights in the development workflow to enhance your efficiency. Meanwhile, they request some additional responsibilities from you:
```
Comment (tisonkun) [inline src/contribute-to-tidb/community-guideline.md:18]:
```suggestion
A typical promoted path for a TiDB developer is from user to reviewer, then committer and maintainer, finally maybe a TOC member. But gaining more roles doesn't mean you have any privilege over other community members or even any right to control them. Everyone in TiDB community are equal and share the responsibility to collaborate constructively with other contributors, building a friendly community. The roles are a natural reward for your substantial contribution in TiDB development and provide you more rights in the development workflow to enhance your efficiency. Meanwhile, they request some additional responsibilities from you:
```

File: src/contribute-to-tidb/community-guideline.md [hunk 5]
```diff
@@ -1,31 +1,24 @@
...
+* Committer/maintainer have the right to merge pull requests, so bear the additional responsibility of handling the consequences of accepting a change into the code base or documentation. That includes reverting or fixing it if it causes problems as well as helping out the release manager in resolving any problems found during the pre-release testing cycle. While all contributors are free to help out with this part of the process, and it is most welcome when they do, the actual responsibility rests with the committers/maintainers that merged the change.
+* Review/committer/maintainer also bear the primary responsibility for guiding contributors the right working procedure, like deciding when changes proposed on the issue tracker should be escalated to [internal.tidb.io](https://internals.tidb.io) for wider discussion, as well as suggesting the use of the [TiDB Design Documents](https://github.com/pingcap/tidb/tree/master/docs/design) process to manage the design and justification of complex changes, or changes with a potentially significant impact on end users.
```
Comment (tisonkun) [inline src/contribute-to-tidb/community-guideline.md:22]:
```suggestion
* Now that you are a member of team reviewers/committers/maintainers, you are representing the project and your fellow team members whenever you discuss TiDB with anyone. So please be a good person to defend the reputation of the team.
* Committers/maintainers have the right to merge pull requests, so bear the additional responsibility of handling the consequences of accepting a change into the codebase or documentation. That includes reverting or fixing it if it causes problems as well as helping out the release manager in resolving any problems found during the pre-release testing cycle. While all contributors are free to help out with this part of the process, and it is most welcome when they do, the actual responsibility rests with the committers/maintainers that merged the change.
* Reviewers/committers/maintainers also bear the primary responsibility for guiding contributors the right working procedure, like deciding when changes proposed on the issue tracker should be escalated to [internal.tidb.io](https://internals.tidb.io) for wider discussion, as well as suggesting the use of the [TiDB Design Documents](https://github.com/pingcap/tidb/tree/master/docs/design) process to manage the design and justification of complex changes, or changes with a potentially significant impact on end users.
```

**Comment by tisonkun** [comment]
Thanks for your contribution @zhangyangyu ! Several comments inline.

**Review by tisonkun — APPROVED** [review]
LGTM. Thanks for your contribution @zhangyangyu !

Merging...

**Comment by tisonkun** [comment]
@all-contributors please add @zhangyangyu  for content

**Pull request merged**